client/infinispan-key-value-store-hotrod/src/main/java/org/infinispan/api/client/impl/InfinispanClientImpl.java

```diff
@@ -0,0 +1,104 @@
+package org.infinispan.api.client.impl;
+
+import static org.infinispan.query.remote.client.ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+
+import org.infinispan.api.configuration.ClientConfig;
+import org.infinispan.api.Infinispan;
+import org.infinispan.api.reactive.client.impl.KeyValueStoreImpl;
+import org.infinispan.api.exception.InfinispanException;
+import org.infinispan.api.marshalling.Marshaller;
+import org.infinispan.api.reactive.KeyValueStore;
+import org.infinispan.api.reactive.KeyValueStoreConfig;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller;
+import org.infinispan.protostream.BaseMarshaller;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
+
+public class InfinispanClientImpl implements Infinispan {
+
+   private RemoteCacheManager cacheManager;
+   private Set<String> files = ConcurrentHashMap.newKeySet();
+   private ExecutorService asyncExecutorService;
+
+   public InfinispanClientImpl(ClientConfig clientConfig) {
+      if (clientConfig instanceof ClientConfigurationLoader.ConfigurationWrapper) {
+         ClientConfigurationLoader.ConfigurationWrapper wrapper = (ClientConfigurationLoader.ConfigurationWrapper) clientConfig;
+         cacheManager = new RemoteCacheManager(wrapper.getConfiguration());
+      }
+      asyncExecutorService = cacheManager.getAsyncExecutorService();
```

Comment:
may throw a `NullPointerException` if `clientConfig` is something else...

Reply:
adding an exception